feat: JWT 인증 기반 팔로우 기능 구현

dain391 opened the pull request:
## 이슈 번호
#39

## 작업 내용
- [x] 컨트롤러에서 userId를 파라미터로 받지 않고 JWT 기반으로 인증 처리
- [x] JwtUtil을 활용해 HTTP 요청 헤더의 JWT에서 userId 추출
- [x] 서비스 메서드명 및 반환 타입 수정 및 일관성 유지
- [x] 불필요한 중복 제거 및 예외 처리 개선
- 
## 스크린샷(선택)

dain391 committed:
fix: FollowService에서 JwtUtil 의존성 제거 및 API 응답 형식 통일

- JwtUtil 클래스 의존성을 FollowService에서 제거하고, Controller에서 @RequestAttribute(RequestAttributeKey.USER_ID)로 사용자 ID를 받도록 변경
- 시크릿 키 오류 해결

resolves: #39

GyeongSe99 commented on newsfeed/src/main/java/com/npcamp/newsfeed/follow/controller/FollowController.java:
```diff
@@ -1,45 +1,56 @@
-    public ResponseEntity<ApiResponse<FollowResponseDto>> createFollow(@Valid @RequestBody FollowRequestDto followDto) {
+    public ResponseEntity<ApiResponse<FollowResponseDto>> createFollow(
+            @RequestAttribute("USER_ID") Long followerUserId,
+            @RequestBody FollowRequestDto followDto
```
@Valid 추가해주세요!

GyeongSe99 replied:
@RequestAttribute(RequestAttributeKey.USER_ID) 로 바꿔주세요!

GyeongSe99 commented on newsfeed/src/main/java/com/npcamp/newsfeed/follow/controller/FollowController.java:
```diff
@@ -1,45 +1,56 @@
-    public ResponseEntity<ApiResponse<List<FollowResponseDto>>> getFollowers(@PathVariable Long followeeId) {
-        List<FollowResponseDto> followers = followService.getFolloweesByFollower(followeeId);
-        return ResponseEntity.ok(ApiResponse.success(followers));
+    // 단일 팔로우 조회
+    @GetMapping("/{id}")
+    public ResponseEntity<ApiResponse<Follow>> getFollowById(@PathVariable Long id) {
+        Follow follow = followService.getFollowById(id);
+        return new ResponseEntity<>(ApiResponse.success(follow), HttpStatus.OK);
```
다른 추가적인 값들을 리턴하도록 수정할 수 있기 때문에 확장성을 위해 Dto 로 감싸 반환해주세요

GyeongSe99 commented on newsfeed/src/main/java/com/npcamp/newsfeed/follow/service/FollowService.java:
```diff
@@ -8,44 +8,42 @@
-                .collect(Collectors.toList());
+    public List<Follow> getFollowings(Long userId) {
+        return followRepository.findByFollowerUserId(userId);
     }
```
이전처럼 Dto로 반환하도록 해주세요

GyeongSe99 commented on newsfeed/src/main/java/com/npcamp/newsfeed/follow/service/FollowService.java:
```diff
@@ -8,44 +8,42 @@
+                .orElseThrow(() ->
+                        new IllegalArgumentException("해당 팔로우 관계를 찾을 수 없습니다. (id: " + id + ")")
+                );
     }
```
ResourceNotFoundException으로 예외처리해주세요!

GyeongSe99 replied:
Dto로 반환하도록 해주세요!

dain391 committed:
feat: 팔로우 생성 시 followeeUserId를 DTO에서 추출하도록 수정

- @Valid 추가

resolves: #39

dain391 committed:
refactor: 팔로우 목록 조회 시 엔티티 대신 Dto로 반환하도록 리팩토링

- 서비스 계층에서 Follow → FollowResponseDto로 변환하여 반환
- 컨트롤러에서도 List<Follow> → List<FollowResponseDto>로 리턴 타입 변경
- 응답 구조 확장성을 위한 설계 개선

resolves: #39

dain391 committed:
fix: 예외 처리 및 DTO 반환 수정

- 팔로우 단건 조회 시 ResourceNotFoundException으로 예외 처리
- 팔로우 단건 조회 시 DTO 반환으로 수정

resolves: #39

dain391 committed:
fix: 팔로우 생성 오류 수정

- FollowRequestDto의 followerUserId를 삭제하여 제대로 입력해도 빈 것으로 인식되는 오류 수정

resolves: #39

dain391 committed:
refactor: 컨트롤러 클래스 수정

- JwtUtil 대신 @RequestAttribute를 사용해 사용자 ID 추출 방식 개선

resolves: #39
